### PR TITLE
[MISC] Deprecate views::take_line

### PR DESCRIPTION
**Pull request by SGSSGene**
solves https://github.com/seqan/product_backlog/issues/359

- `seqan3::views::take_line` is moved to `seqan3/io/detail/views` for
internal use only. A deprecated version of `take_line` is kept in its
original place.
- moved all snippets accordingly
- moved test accordingly
- add CHANGELOG entry

Note: this PR differs slightly from the Issue. This PR doesn't remove test/snippets, it just moves them accordingly.

### CODE REVIEW

File: include/seqan3/io/detail/views/take_line.hpp [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+#pragma once
+
+#include <seqan3/range/views/take_until.hpp>
+#include <seqan3/utility/char_operations/predicate.hpp>
+
+// ============================================================================
+//  io::detail::take_line (adaptor instance definition)
+// ============================================================================
+
+namespace seqan3::io::detail
```
Comment (marehr) [inline include/seqan3/io/detail/views/take_line.hpp:22]:
```suggestion
namespace seqan3::detail
```

**Comment by marehr** [comment]
```
/home/runner/work/seqan3/seqan3/seqan3/include/seqan3/io/detail/views/take_line.hpp:84: warning: @copydetails or @copydoc target 'seqan3::io::detail::take_line' not found
```

File: include/seqan3/io/sequence_file/format_fasta.hpp [hunk 2]
```diff
@@ -247,7 +247,7 @@ class format_fasta
 
             #else // ↑↑↑ WORKAROUND | ORIGINAL ↓↓↓
 
-                std::ranges::copy(stream_view | views::take_line_or_throw                    // read line
+                std::ranges::copy(stream_view | detail::take_line_or_throw                    // read line
                                               | std::views::drop_while(is_id || is_blank)    // skip leading >
```
Comment (Irallia) [inline include/seqan3/io/sequence_file/format_fasta.hpp:251]:
```suggestion
                std::ranges::copy(stream_view | detail::take_line_or_throw                   // read line
                                              | std::views::drop_while(is_id || is_blank)    // skip leading >
```

File: include/seqan3/io/structure_file/format_vienna.hpp [hunk 3]
```diff
@@ -174,8 +174,8 @@ class format_vienna
         if constexpr (!detail::decays_to_ignore_v<seq_type>)
         {
             auto constexpr is_legal_seq = char_is_valid_for<seq_legal_alph_type>;
-            std::ranges::copy(stream_view | views::take_line_or_throw                      // until end of line
-                                          | std::views::filter(!(is_space || is_digit)) // ignore whitespace and numbers
+            std::ranges::copy(stream_view | detail::take_line_or_throw               // until end of line
+                                          | std::views::filter(!(is_space || is_digit))  // ignore whitespace and numbers
```
Comment (Irallia) [inline include/seqan3/io/structure_file/format_vienna.hpp:178]:
```suggestion
            std::ranges::copy(stream_view | detail::take_line_or_throw                   // until end of line
                                          | std::views::filter(!(is_space || is_digit))  // ignore whitespace and numbers
```

**Review by Irallia — APPROVED** [review]
LGFM,
the only thing I saw, were two comments that are not aligned anymore. 💅

**Comment by SGSSGene** [comment]
> LGFM,
> the only thing I saw, were two comments that are not aligned anymore. nail_care

Yes, polishing a little annoying but important!

**Review by marehr — APPROVED** [review]
Only issue is that range/views/all.hpp does not define `seqan3::views::take_line` etc anymore

We could solve this be defining those entities in `include/seqan3/io/detail/take_line_view.hpp`, like you did in your other PR.

**Pull request merged**